### PR TITLE
Upgrade alpine to 3.16.0

### LINKED ISSUE #1990: Update to alpine 3.16 images
#### Is your feature request related to a problem? Please describe.

We're currently basing Mimir images off of 3.15.0 which has had a few patch releases so far (latest at time of writing is 3.15.4)

#### Describe the solution you'd like

Update to `alpine:3.16.0` in each of our Docker images.

#### Describe alternatives you've considered

N/A

#### Additional context

Our internal security scanner has found a variety of issues with tools in 3.15 (unrelated to Mimir itself).

### PR DESCRIPTION
**Pull request by williamzelesny**
#### What this PR does

Upgrades alpine from 3.15.0 to 3.16.0

#### Which issue(s) this PR fixes or relates to

Fixes #1990

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -773,6 +773,7 @@ _Changes since Cortex 1.10.0._
 * [ENHANCEMENT] Overrides exporter: Add `max_fetched_chunks_per_query` and `max_global_exemplars_per_user` limits to the default and per-tenant limits exported as metrics. #471 #515
 * [ENHANCEMENT] Upgrade Go to 1.17.8. #1347 #1381
 * [ENHANCEMENT] Upgrade Docker base images to `alpine:3.15.0`. #1348
+* [ENHANCEMENT] Upgrade Docker base images to `alpine:3.16.0`. #1990
```
Comment (pracucci) [inline CHANGELOG.md:776]: Please move this CHANGELOG entry above, under the `main / unreleased` section.

Comment (pracucci) [reply CHANGELOG.md:776]: Also please reference the PR number #2028 instead of #1990.

**Review by pracucci — COMMENTED** [review]
Thanks for your contribution!

Could you also look for `alpine::3.15` across the code base, please? There are few places where we currently use a different patch version and we should update them too.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## Grafana Mimir - main / unreleased
+* [ENHANCEMENT] Upgrade Docker base images to `alpine:3.16.0`. #2028
```
Comment (pracucci) [inline CHANGELOG.md:4]: Sorry, I think I didn't myself clear enough. Previously, the CHANGELOG entry was around line 700, in the CHANGELOG section of a previous version. You should move this to entry to line 29, at the end of the current `[ENHANCEMENT]` block, related to the "main / unreleased" changes.

**Review by pracucci — COMMENTED** [review]
Thanks for the follow up! A nit and then it should be good to go!

File: CHANGELOG.md (hunk 2 above)
Comment (williamzelesny) [reply CHANGELOG.md]: my fault, I see it now, serves me right for making changes prior to having coffee. Thanks for the help reviewing!

File: CHANGELOG.md [hunk 3]
```diff
@@ -29,6 +29,7 @@
 * [BUGFIX] Ingester: fixed deceiving error log "failed to update cached shipped blocks after shipper initialisation", occurring for each new tenant in the ingester. #1893
 * [BUGFIX] Ring: fix bug where instances may appear unhealthy in the hash ring web UI even though they are not. #1933
 * [BUGFIX] API: gzip is now enforced when identity encoding is explicitly rejected. #1864
+* [ENHANCEMENT] Upgrade Docker base images to `alpine:3.16.0`. #2028
```
Comment (pracucci) [inline CHANGELOG.md:32]: Don't hate me, please. As described in the contributing guide https://github.com/grafana/mimir/tree/main/docs/internal/contributing#changelog , the CHANGELOG entries are grouped togehter. This entry should be put at the end of the `[ENHNACEMENT]` entries group, right after line 28.

File: CHANGELOG.md [hunk 4]
```diff
@@ -14,17 +14,18 @@
     - `-querier.query-store-after`
 * [CHANGE] Config flag category overrides can be set dynamically at runtime. #1934
 * [CHANGE] Ingester: deprecated `-ingester.ring.join-after`. Mimir now behaves as this setting is always set to 0s. This configuration option will be removed in Mimir 2.4.0. #1965
+* [CHANGE] Blocks uploaded by ingester no longer contain `__org_id__` label. Compactor now ignores this label and will compact blocks with and without this label together. `mimirconvert` tool will remove the label from blocks as "unknown" label. #1972
```
Comment (pracucci) [inline CHANGELOG.md:17]: Thanks for fixing it! I didn't notice it.

**Review by pracucci — APPROVED** [review]
Thanks!

**Pull request merged**